### ISSUE TITLE
Could you please release the training dataset?

### ISSUE THREAD

**Issue by felixfuu** [MS-Diffusion/MS-Diffusion#11]
Could you please release the training dataset?

**Comment by sherrydoge** [comment]
We used an internal dataset for training, which we are not authorized to release in accordance with company regulations.

**Comment by felixfuu** [comment]
@sherrydoge  Can you release the 2.8M general scenario videos datasets?

**Comment by sherrydoge** [comment]
I'm sorry that both the general scenario and the product demonstration videos are internal. You can refer to Appendix A in the paper for the data construction pipeline.

**Comment by felixfuu** [comment]
@sherrydoge 
There are several questions about the video data:
1. How to collect these video data?
2. How long does it usually take to take a frame of video? Because adjacent frames are almost the same;
3. If the overlap between objects is very serious, how to solve this problem? should we just throw it away?
4. We have obtained the mask by SAM, why do we use the box in the `Resampler` instead of the mask?

**Comment by sherrydoge** [comment]
1. We use collected internal datasets
2. The videos are split into several clips, and a fixed number of frames are extracted with the same interval
3. We also use these data for training
4. It is hard to set a mask when inferencing

**Comment by felixfuu** [comment]
@sherrydoge 
1.  What is the role of `need_text_attention_map` : https://github.com/MS-Diffusion/MS-Diffusion/blob/f7e470b1ae0929233a9f3641609895673a5eb172/msdiffusion/models/attention_processor.py#L130
2. Similarly, what does `use_psuedo_attention_mask` do? https://github.com/MS-Diffusion/MS-Diffusion/blob/f7e470b1ae0929233a9f3641609895673a5eb172/msdiffusion/models/attention_processor.py#L133C14-L133C40
3. And `mask_threshold`: https://github.com/MS-Diffusion/MS-Diffusion/blob/f7e470b1ae0929233a9f3641609895673a5eb172/msdiffusion/models/model.py#L133C12-L133C27

**Comment by sherrydoge** [comment]
Please refer to https://github.com/MS-Diffusion/MS-Diffusion?tab=readme-ov-file#auto-layout-guidance

**Comment by felixfuu** [comment]
@sherrydoge  thx ~

**Comment by sherrydoge** [comment]
1. no. It's only a feature that can be used in the inference
2. We pad/discard the ref images to 4 in the training. You can also set another number based on your dataset.

**Comment by felixfuu** [comment]
@sherrydoge 
two more questions:
1. During the training process, will the cropped ref images be processed by mask(get by sam) or used directly?
2. Every iter the pad 4 images, are they directly filled with 0, and the box is also 0?

**Comment by sherrydoge** [comment]
1. yes, they are masked before training
2. yes, the box is set to 0 so that they can be masked in the cross-attention

**Comment by felixfuu** [comment]
@sherrydoge I used 1 million data to try to reproduce the results(1w iter, about 160w data), but I didn't see any similar adapter results. What is the approximate size of the box that needs to be filtered out during training?

**Comment by felixfuu** [comment]
@sherrydoge  I used 1 million data to try to reproduce the results(1w iter, about 160w data), but I didn't see any similar adapter results.
What is the approximate size of the box that needs to be filtered out during training?

**Comment by felixfuu** [comment]
@sherrydoge Can you give your experimental thr?  aspect ratio, min width/height, max width/heigh
